 design/README.MD

```diff
@@ -0,0 +1,456 @@
+## Overview
+
+Smithy is an interface definition language.
+It lets you abstract your interface
+from any given executable implementation.
+By focusing on the abstract interface
+Smithy helps you build an interface
+that can safely evolve
+maintaining both forward and backwards compatibility.
+
+Dafny is a programming language with formal verification features.
+Dafny code is translated to higher-level languages
+to provide interoperability for your verified code
+like Java or C#.
+These native representations are then executable.
+This means that a single Dafny program
+can be translated into multiple interoperable programs.
+
+Dafny itself has a `string` type
+that you might assume can map
+to the obvious string type in each translated language.
+
+However a native `string` may be UTF8, UTF16/USC2, or UTF32 
+depending on the language.
+Numbers are similarly complicated.
+Some languages have different sized numbers,
+while others have either a different default or a single size.
+There are also complicated types like streams.
+All these languages have equivalent
+but not congruent types to each other.
+
+In this project Dafny and Smithy are combined
+to generate concrete interfaces in Dafny
+and supported Dafny translation targets.
+These generated interfaces all use the same translated Dafny implementation
+and expose native interfaces and types for each language.
+
+The Smithy shapes are the common denominator
+between all these different types/conversions.
+
+Concretely our traditional greeting program:
+`() => "Hello World!"`
+generate interfaces that can be used like this:
+1. Dafny -> `var hello: seq<char> :- greet();`
+1. Java -> `String hello = greet();`
+1. C# -> `string hello = greet();`
+1. Python -> `hello = greet()`
+1. Rust -> `let hello: String = greet();`
+
+## Definitions
+
+- Runtime or native runtime:
+    A Dafny translation target language,
+    e.g Java or C#.
+    This also may include additional context
+    like language version or other versions/features.
+- Native type or native interface:
+    A type/interface in a runtime,
+    e.g. Java `String`
+- Smithy-runtime:
+    The project used to generate native types
+    for Smithy shapes.
+- Polymorph or polymorph-runtime:
+    The transformation layer between Dafny types
+    and native types,
+    e.g. polymorph-java.
+- Shim:
+    The object that exports the native interface
+    and connects to the internal Dafny implementation.
+- Type converters, ToDafny, ToNative:
+    The dual functions that convert between
+    Dafny and native types.
+    e.g. `D == ToDafny(ToNative(D))`,
+    `N == ToNative(ToDafny(N))`
+
+### Conventions used in this document
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL"
+in this document are to be interpreted as described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+## Smithy-Dafny modes
+
+### Library
```

Comment:
is the term "library" too generic here? if this is specifically about `@localService` then maybe call the section `Local Service` since, from what i understand, this is about 100% client side code with no HTTP requests. or, at least, an HTTP request is not the sole abstraction like with AWS service SDKs.